book_my_ticket/sources/book_my_ticket.move

```diff
@@ -194,45 +241,104 @@ module TicketProject::BookMyTicket {
             ticket_type
         });
+        
     }
+
+
+    public entry fun claim_ticket(self : TicketNFT , _ : &mut TxContext){
+        let TicketNFT{ id ,ticket_type ,description,ticket_id ,ticket_claimed} = self ;
+        object::delete(id);
+        
+        emit(TicketClaimedEvent{
+            ticket_type,
+            ticket_id,
+            description ,
+            ticket_claimed  
+            })
+    }
+
+    // public entry fun add_admin(owner_cap : OwnerCap ,admin_addr : address ,ctx : &mut TxContext ){
+
+    // }
     
     /// Adds new ticket types to the BookMyTicket platform.
     /// This function allows the platform owner to add new ticket types along with their prices.
-    public entry fun add_ticket_types(platform_info: &mut BmtPlatformDetails, ticket_type: vector<String>, price: vector<u64>, ctx: &mut TxContext) {
-        let sender: address  = tx_context::sender(ctx);
+    public entry fun add_ticket_types(_ :&OwnerCap , platform_info: &mut BmtPlatformDetails, ticket_type: vector<String>,ticket_price : vector<u64> , _ctx: &mut TxContext) {
+        // let sender: address  = tx_context::sender(ctx);
         let type_len: u64 = vector::length(&ticket_type);
-        let price_len: u64 = vector::length(&price);
-        assert!(sender == platform_info.owner, EINVALID_OWNER);
-        assert!(type_len == price_len, EINVALID_LENGTH);
+        let price_len: u64 = vector::length(&ticket_price);
+        assert!(type_len == price_len , EINVALID_LENGTH);
         
-        let temp_ticket_type: &mut Table<String, u64> = &mut platform_info.ticket_types;
+        let temp_ticket_type: &mut VecMap<String, u64> = &mut platform_info.ticket_types;
 
         while (!vector::is_empty(&ticket_type)) {
             let ticket_type: String = vector::pop_back(&mut ticket_type);
-            let ticket_price: u64  = vector::pop_back(&mut price);
+            let ticket_price: u64  = vector::pop_back(&mut ticket_price);
 
-            event::emit(TicketTypeAdded {
+            emit(TicketTypeAddedEvent {
                 ticket_type,
                 ticket_price
             });
 
-            table::add(temp_ticket_type, ticket_type, ticket_price)
+            vec_map::insert(temp_ticket_type, ticket_type, ticket_price)
+        }
+    }
+
+
+    public entry fun remove_ticket_type(_ : &OwnerCap , platform_info: &mut BmtPlatformDetails, ticket_type: vector<String> , _ctx: &mut TxContext){
+            // let sender: address  = tx_context::sender(ctx);
+        let type_len: u64 = vector::length(&ticket_type);
+        assert!(type_len > 0 , EINVALID_LENGTH);
+        
+        let temp_ticket_type: &mut VecMap<String, u64> = &mut platform_info.ticket_types;
+
+        while (!vector::is_empty(&ticket_type)) {
+            let ticket_type: String = vector::pop_back(&mut ticket_type);
+
+            let (ticket_type ,ticket_price ) = vec_map::remove(temp_ticket_type, &ticket_type);
```

Comment:
before removing it, I think you need to check even `ticket_type` exists in the vecMap or not.
what if the `ticket_type` which comes in the parameter does not exist? Does transaction abort?

Reply:
Aborts if `key` is not bound in `self`.  this feature is inbuilt in the vec_map itself